### PR TITLE
fix: windows GL error message handling

### LINKED ISSUE #286: set_background_color error on windows machine
Running any function from visuals (set_background_color, FixationDot, etc) on windows results in the following error:

```
Traceback (most recent call last):
  File "C:\Experiments\mbagheri\aud_att_int_SAMPLE\aud_att_int.py", line 35, in <module>
    ec.set_background_color([0.2] * 3)
  File "C:\Experiments\mbagheri\aud_att_int_SAMPLE\expyfun\_experiment_controller.py", line 540, in set_background_color
    Rectangle(self, pos=[0, 0, 2.1, 2.1], fill_color=color).draw()
  File "C:\Experiments\mbagheri\aud_att_int_SAMPLE\expyfun\visual\_visual.py", line 476, in __init__
    line_loop=True)
  File "C:\Experiments\mbagheri\aud_att_int_SAMPLE\expyfun\visual\_visual.py", line 191, in __init__
    _check_log(vertex, gl.glGetShaderInfoLog)
  File "C:\Experiments\mbagheri\aud_att_int_SAMPLE\expyfun\visual\_visual.py", line 172, in _check_log
    raise RuntimeError(message)
RuntimeError: No errors.
``` 
Updating pyglet didn't help. Any ideas?

### PR DESCRIPTION
**Pull request by drammock**
closes #286

### CODE REVIEW

**Comment by drammock** [comment]
Travis is mad because of unrelated Flake errors (when did those slip in?) and because of [a py3k string/bytes problem](https://travis-ci.org/LABSN/expyfun/jobs/147350277#L827).  Will try to iron these out tomorrow morning, if someone else doesn't get to it first.

**Comment by Eric89GXL** [comment]
Restart the builds now it should work again:

https://gitlab.com/pycqa/flake8/issues/177#note_13316550

**Comment by Eric89GXL** [comment]
...although I did restart and still saw errors, so I commented again:

https://gitlab.com/pycqa/flake8/issues/177#note_13318620

**Comment by drammock** [comment]
@Eric89GXL travis fails are now all flake-related, so I think this is ready to merge.

File: expyfun/visual/_visual.py [hunk 1]
```diff
@@ -168,7 +168,9 @@ def _check_log(obj, func):
     ptr = cast(pointer(log), POINTER(c_char))
     func(obj, 4096, pointer(c_int()), ptr)
     message = log.value
-    if message:
+    if message.startswith(b'No errors'):
+        pass
+    elif message:
```
Comment (Eric89GXL) [inline expyfun/visual/_visual.py]: why not just `if not message.startswith(b'No errors')`?

Comment (Eric89GXL) [inline expyfun/visual/_visual.py]: ...and actually, to be backward compatible you almost certainly need to do `if message and not message.startswith(b'No errors'):`

Comment (Eric89GXL) [inline expyfun/visual/_visual.py]: (compared to what I had anyway)

**Comment by Eric89GXL** [comment]
This should work and it's a show-stopper so I'll merge, but we should unify those checks at some point

**Comment by Eric89GXL** [comment]
Thanks @drammock

**Comment by drammock** [comment]
I think there is some cross-platform inconsistency here regarding what `message` is.  Testing on linux seems to yield `''` which won't choke on the `.startswith`, so as long as `message` is not `None` we should be fine.  I agree though that your second suggestion is probably better.  @mirbagheri said he came across some info online claiming that `'No errors'` actually means "there were some errors that we've buried in some other log file"...  I've asked him to add a link to that here.

**Comment by mirbagheri** [comment]
According [here](http://webcache.googleusercontent.com/a/30432) the output of the compiler in `glGetShaderInfoLog` or 'glGetProgrammInfoLog' is driver-specific, and `'No error'` is only for Nvidia GPUs.
[Here](https://github.com/google/gxui/issues/1) people talk about this `COMPILE_STATUS` test as a workaround for the inconsistency.

**Comment by drammock** [comment]
@mirbagheri first link is dead.

**Comment by mirbagheri** [comment]
Link edited^

**Comment by Eric89GXL** [comment]
So @mirbagheri we should be safe with the now-implemented check?

**Pull request merged**